new design for v3

flynnduism opened the pull request:
This PR refreshes the design of the helm.sh site for V3, to tie into the updated [branding](https://github.com/cncf/artwork/blob/master/examples/incubating.md#helm-logos) and to freshen up the look and feel of the project. 

## Main changes:

* changes the underlying CSS framework from foundation to bulma
* higher contrast for better legibility (no more waves or white text on blue backgrounds) 
* reworking the menus, esp for mobile
* deleting lots of old code

flynnduism commented:
Preview here: https://deploy-preview-346--helm-merge.netlify.com/

- [ ] mobile sizing adjustments
- [ ] minificaton
- [ ] homepage cleanups
- [ ] blog restore template grid columns

bacongobbler commented:
Wow. The redesign looks amazing!

When this is ready for a code review, can you un-mark this as a draft? Let me know ready for a full review.

flynnduism commented:
> When this is ready for a code review, can you un-mark this as a draft? Let me know when you're ready for a full review.

Thanks @bacongobbler. It's _almost almost_ there.

<img width="193" alt="Screen Shot 2020-01-28 at 10 32 20 AM" src="https://user-images.githubusercontent.com/686194/73289198-84251780-41b9-11ea-8bf0-4819950d5f34.png">

There are two significant CSS/layout bugs that need resolving before this will be ready to review. I will be working on these this afternoon, so I hope to switch this to be reviewable tonight.

---

<img width="1565" alt="Screen Shot 2020-01-28 at 10 31 03 AM" src="https://user-images.githubusercontent.com/686194/73289082-4e802e80-41b9-11ea-90ff-8bc5fbd65c68.png">

Blog grid layout needs fixing. 

<img width="1608" alt="Screen Shot 2020-01-28 at 10 31 22 AM" src="https://user-images.githubusercontent.com/686194/73289133-65268580-41b9-11ea-8d67-4c539c4fff1b.png">


Homepage features need to be tidied.

flynnduism commented:
Ready for some 👀 to review these changes. Preview [new site look here](https://deploy-preview-346--helm-merge.netlify.com/).

florianwiech commented:
some feedback from my side:
At first I thought why helm needs a new look, because the old side is just fine. But when I compared the two websites I have to say kudos to @flynnduism , the new design looks great! :confetti_ball:
So thank you @flynnduism for taking the time and redesigning the page!
The aspect of using the current theme for [Helm v2](https://v2.helm.sh) and refresh the website style for Helm 3 is a great idea. Developers directly know which version they are looking at.

Here are some issues I found:
- GitHub logo missing; The "Get Helm" button links to GitHub, but I think it is a bit unintuitive
- The blue border around the complete page layout looks great. Unfortunately it covers the scroll bar on macOS devices, that it is hard to see the current scroll position (maybe a light scroll bar could solve the problem)
- Landing page
  - Hero:
    - Helm logo is linked to the same site I am on. Is this link necessary?
    - The "Kubernetes" link is not clickable, because the ship animation with the "ahoy" badge covers the link (try inspecting it in a browser with the development tools)
  - The ship logo does not move in a smooth way. (The animations of the waves on the other hand are great!)
  - In the blue background area, the "Learn more" list is not that visible. The links and list item dots are not really visible. (maybe colouring the dots in white; underlining the links or cover them in the orangish color that is used in other parts of the site)
  - The "Upgrade Helm v2 to v3" Banner needs the link to the upgrade guide
  - The "More information about the Helm project, and how to contribute." heading is a bit off (the current helm site has the same issue, but the text fonts differ not that much, so it is not that visible there). It can be solved by adding the following style: `.helm-community > p { padding-right: 0px; }`
  - It would be great, if the links in the section "Developer Standups" are underlined by default
- Docs code fragments
  - I think the light orange border style is a bit misleading here. Orange and red are for warnings and errors. So someone could interpret that all the code snippets shown are wrong. I would prefer a blueish or blackish border color.
- 404 page
  - Navigation shows two helm logos and is a bit of when scrolling
  - the "report it here" link is covered like the "Kubernetes" link on the landing page

flynnduism commented:
Thanks @florianwiech and @hickeyma for looking this over. These comments were super thorough and thoughtful - thank you so much for taking the time to go through the site and review it!

I've run through the theme again and made a bunch of changes (and _hopefully_ improvements)  as suggested.

---


### Changed:

##### **Get Helm button**
You both queried the clarity and usefullness of this. This button should be the most direct way to go and dive into Helm - to start installing and using. Instead of going to the Github repo _(from one landing to another)_ I've changed it to the [Quickstart Guide](https://helm.sh/docs/intro/quickstart/) doc.

##### Fixed Links
* Homepage: kubernetes.io link in the intro
* Homepage: link to [migration guide](https://helm.sh/docs/topics/v2_v3_migration/) in v3 section
* Homepage: learn more links styled to be clearer

##### Blue border obscures the scroll position
I've the way background color is inherited so that the default OS scrollbars are inverted to a light color, making them visible against the page border.

##### _'for'_ Kubernetes
I've removed the italics on `for`. The intention was to add some design flair to the headline but it doesn't make grammatical sense. 

##### Code Snippets
Removed the orange-y border and instread use neutral styling.

##### 404 Page
Fixed glitches on this layout.

##### Copy edits
@hickeyma I applied all of your  suggested edits to the homepage, much better!

##### Community Info
I went beyond the comments regarding the formatting of this area and made some changes to highlight contribution info and add an Events area.

<img width="1553" alt="Screen Shot 2020-02-27 at 3 32 25 PM" src="https://user-images.githubusercontent.com/686194/75496555-60025500-5976-11ea-9988-febfb2c41fe3.png">

---

### Not Changed:


##### Homepage: boat animation
I've tweaked this to simplify the transitions involved. However there is still a slow _'shake'_ in the svg image as it moves. The browser rendering is not as smooth as I would like.

The boat only appears on the landing page with the aim of adding some fun and personality to Helm. I'd still like to include it in the redesign, but will add a follow-up issue to optimize this further.

##### Homepage: Helm triangle
> The Helm triangle image above "What is Helm?" looks clipped

I'm not sure aobout this point. Do you mean the way the edges of Helm sit in the triangle _(highlighted below)_? Or is part of the triangle cut off?

![cropped-helm-triangle](https://user-images.githubusercontent.com/686194/75473689-80b4b580-594a-11ea-9206-f40399282e1e.png) 

---

Preview site here: https://deploy-preview-346--helm-merge.netlify.com/

hickeyma commented:
@flynnduism Thanks for the updates, it looks great. 

> Do you mean the way the edges of Helm sit in the triangle (highlighted below)? 

Yeah, the highlighted bits you show above. Looking at it again it is because the Helm writing is in the same colour as the background. I think its ok as is. 

Overall, LGTM 👍

flynnduism commented:
Thanks @hickeyma and @jdolitsky for approval!

lucperkins commented on config.toml:
```diff
@@ -26,115 +26,122 @@ disableKinds = ["taxonomy", "taxonomyTerm"]
+  languageName = "English"
+  weight = 1
+
+    [[languages.en.menus.main]]
```
Nit: TOML is designed around the principle that indentation is not necessary. No need to change in this PR but I'd advise removing this unnecessary complexity.

lucperkins commented:
@flynnduism Could you rebase this against `master`? We should make sure that the new theme goes through the link check gauntlet 👿

flynnduism commented:
Thanks @lucperkins, I've updated as recommended.
